docs: estrutura Wiki de tutoriais e configuração de APIs de IA

Rossi-Luciano opened the pull request:
## O que esse PR faz?

Este PR implementa melhorias em documentação e configuração do sistema:

1. **Documentação estruturada via Wiki**:
   - Adiciona página principal com visão geral do sistema e público-alvo
   - Cria fluxograma visual do processo com legenda explicativa
   - Estrutura tutoriais organizados (Instalação, Processamento, Validação)
   - Padroniza terminologia ("conversão" → "processamento")

2. **Configuração de APIs de IA**:
   - Adiciona suporte a Hugging Face (HF_TOKEN)
   - Habilita integração com Llama (LLAMA_ENABLED)
   - Configura Google Gemini API (GEMINI_API_KEY)

3. **Padronização de código**:
   - Ajusta formatação PEP8 em `xml_manager/models.py`
   - Corrige quebras de linha em ForeignKeys
   - Remove espaços em branco desnecessários

## Onde a revisão poderia começar?

**Documentação Wiki:**
- Página principal (Home)
- Fluxograma do processo (verificar renderização Mermaid)
- Links entre páginas de tutoriais
- Consistência terminológica em todas as páginas

**Arquivos de código:**
- `/.envs/.local/.django`
  - Validar se as chaves de API devem estar versionadas (considerar usar secrets)
  - Verificar configuração padrão `LLAMA_ENABLED=True`
- `/xml_manager/models.py`
  - Classes `XMLDocument`, `XMLDocumentPDF`, `XMLDocumentHTML`
  - Métodos `create()` reformatados

## Como este poderia ser testado manualmente?

**Documentação:**
1. Acessar Wiki do repositório
2. Navegar pela estrutura de páginas
3. Testar todos os links internos
4. Verificar renderização do fluxograma Mermaid

**Código:**
1. Verificar carregamento das variáveis de ambiente:
```bash
   python manage.py shell
   >>> from django.conf import settings
   >>> print(settings.HF_TOKEN)
   >>> print(settings.LLAMA_ENABLED)
   >>> print(settings.GEMINI_API_KEY)
```

2. Executar testes existentes:
```bash
   python manage.py test xml_manager
```

## Algum cenário de contexto que queira dar?

**Documentação:**
- Desenvolvida para público-alvo com diferentes níveis técnicos
- Estrutura segue fluxo natural: Instalação → Processamento → Validação
- Terminologia padronizada para facilitar compreensão

## Quais são tickets relevantes?

NA

## Referências

- Padrão JATS/SciELO PS
- [Hugging Face API Tokens](https://huggingface.co/docs/hub/security-tokens)
- [Google AI Gemini API](https://ai.google.dev/gemini-api/docs)
- [PEP 8 Style Guide](https://pep8.org/)

eduranm committed:
Integración de ArticleViewSet en el enrutador y actualización de URLs principales

eduranm committed:
Actualización de settings: cambio en directorio de modelos, inclusión de nuevas apps y aumento del límite de campos

eduranm committed:
Ampliación de core.models: nuevos modelos de género, idioma, licencias, textos con idioma y manejo flexible de fechas

eduranm committed:
Actualización en Reference: uso de ReferenceStatus y ajuste en creación de referencias

eduranm committed:
Refactor en reference.config: serialización de ejemplos con json.dumps y ampliación de tipos soportados (confproc, full_text, etc.)

eduranm committed:
Refactor en marker: actualización de importación de GenericLlama desde model_ai

eduranm committed:
Eliminación de reference.tasks (responsabilidad movida a data_utils y model_ai)

eduranm committed:
Refactor en wagtail_hooks: uso de SnippetViewSet y soporte para descarga de modelos

eduranm committed:
Actualización de dependencias: inclusión de langid, google-generativeai y python-docx

eduranm committed:
Creación y ampliación de la app core: modelos comunes, formularios, vistas de búsqueda, utilidades y hooks de Wagtail

eduranm committed:
Nueva app markup_doc: modelos ArticleDocx, API REST, tareas de marcado, utilidades y hooks de Wagtail

eduranm committed:
Nueva librería markuplib: funciones para procesar DOCX y transformaciones OMML a MathML

eduranm committed:
Nueva app model_ai: integración de LLaMA, funciones genéricas, mensajes de inferencia, tareas y hooks de Wagtail

eduranm committed:
Módulos adicionales en reference: configuración de Gemini y utilidades de procesamiento de datos

eduranm committed:
Consolidación de migraciones en reference: actualización de 0001_initial.py y eliminación de migraciones intermedias

eduranm committed:
Actualización de migración inicial en core_settings: cambio de versión de Django y traducción de verbose_name a inglés

eduranm committed:
Creación de migración inicial en core

eduranm committed:
Eliminación de la app llama3: funcionalidades migradas a model_ai

eduranm committed:
fix(markup_doc): capturar ArticleDocxMarkup.DoesNotExist en generate_xml

Corrige el tipo de excepción para responder 404 cuando el registro no existe.

eduranm committed:
refactor(markuplib): eliminar prints de depuración en extracción de enlaces

Reduce ruido en logs y mantiene la función enfocada a su retorno.

eduranm committed:
style(model_ai): reemplazar bare except por Exception en download_model

Mejora legibilidad y buenas prácticas de manejo de errores.

eduranm committed:
fix(reference): corregir literales y separadores en config_gemini para prompt de referencias

Se agregan comillas a campos textuales y se corrigen comas/keys para evitar errores de parseo del prompt.

eduranm committed:
i18n(reference): envolver labels y help_text con gettext_lazy

Permite traducción de 'Mixed Citation' y 'Rating from 1 to 10'.

eduranm committed:
model_ai: corrige etiqueta de i18n y refuerza unicidad de LlamaModel en save()

eduranm committed:
reference: renombra campo 'estatus' a 'status' y ajusta etiqueta a 'Reference status' (incluye migraciones)

eduranm committed:
reference: elimina import no utilizado en wagtail_hooks.py

pitangainnovare committed:
remove antigo app llama3 e resolve conflito em base.py

pitangainnovare committed:
adiciona instruções para fazer build com suporte a Llama

pitangainnovare committed:
melhora imports

pitangainnovare committed:
Apaga modulo legado llama3

pitangainnovare committed:
Resolve conflitos em generic_llama e aplica correções estruturais:
- function_llama passou a ser LlamaInputSettings em llama.py
- generic_llama passou a ser llama.py com LlamaService

pitangainnovare committed:
Cria input settings para references

pitangainnovare committed:
Padroniza deps

pitangainnovare committed:
Padroniza imports em geral (além de adequar à nova nomenclatura)

pitangainnovare committed:
Adiciona método faltante em marker (por causa de merge)

pitangainnovare committed:
Remove linhas comentadas

pitangainnovare committed:
Inclui AI Model na interace, para poder registrar modelos

pitangainnovare committed:
Adequa uso do serviço Llama em tasks

pitangainnovare committed:
Adiciona migrações

pitangainnovare committed:
Corrige nome de parâmetro type (deve ser mode)

pitangainnovare committed:
Flexibiliza attrs de LlamaService

pitangainnovare committed:
Adiciona alguns FIXME

pitangainnovare committed:
Melhora imports

pitangainnovare committed:
Padroniza nome de metodo que obtem tipo de IA (LLAMA ou GEMINI)

Rossi-Luciano committed:
Corrige o nome do site

Rossi-Luciano committed:
Atualiza os conteúdos de 'locale'

Rossi-Luciano committed:
Correções em 'wagtail_hooks'

Rossi-Luciano committed:
Corrige atributo

Rossi-Luciano committed:
Corrige 'data_utils'

samuelveigarangel commented on markup_doc/sync_api.py:
```diff
@@ -0,0 +1,117 @@
+                    new_journals.append(journal)
+
+                url = data.get("next")
+            except:
+                print('**ERROR url')
+                print(url)
+                url = None
```
Trocar print por logging e inserir uma mensagem mais descritiva do error.

Rossi-Luciano committed:
refactor: reorganiza estrutura de volumes Docker

- Adiciona scielo_xml_tools.yml com novos caminhos de volume
- Move volumes para estrutura ../markup_data/
- Corrige nomes de containers no Makefile (markapi_local_*)
- Adiciona .ipython/ ao .dockerignore
- Adiciona huggingface-hub ao requirements/local.txt
- Atualiza .gitignore para ignorar backups e arquivos temporários
